conf/clusterrunner-master

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:
+# Default-Stop:      0 1 6
+# Short-Description: ClusterRunner init script
+# Description:       Enable clusterrunner service provided by daemon.
+### END INIT INFO
+
+dir="/var/lib/clusterrunner"
+cmd="nohup clusterrunner master"
+user="jenkins"
+
+name=`basename $0`
+pid_file="/var/run/$name.pid"
+stdout_log="/var/log/$name.log"
+stderr_log="/var/log/$name.err"
+
+get_pid() {
+    cat "$pid_file"
+}
+
+is_running() {
+    [ -f "$pid_file" ] && ps -p `get_pid` > /dev/null 2>&1
+}
+
+case "$1" in
+    start)
+    if is_running; then
+        echo "Already started"
+    else
+        echo "Starting $name"
+        cd "$dir"
```

Comment:
Generally not a fan of a process changing the working directory it started off with. How about executing all of this code in a subshell?
```
echo "Starting $name"
(
  cd "$dir"
  if [-z "$user" ]; then
  ...
)
```

Reply:
hmmm, then again, maybe this is a terrible idea. If the subshell exits 1 you'd probably want the whole script to exit 1. There might be some other caveats I haven't thought of. Lets discuss this when you get back from PTO